src/scsiencrypt.h

```diff
@@ -84,6 +86,46 @@ constexpr uint32_t BSLONG(const uint8_t *p)
 #define STENC_BIG_ENDIAN 0
 #endif
 
+// key-associated data
+struct __attribute__((packed)) scsi_kad {
+  uint8_t type;
+  uint8_t flags;
+static constexpr auto flags_authenticated_pos {0u};
+static constexpr uint8_t flags_authenticated_mask {7u << flags_authenticated_pos};
+  uint16_t length;
+  uint8_t descriptor[];
+};
+static_assert(sizeof(scsi_kad) == 4);
+
+// device encryption status page
+struct __attribute__((packed)) scsi_page_des {
+  uint16_t page_code;
+  uint16_t length;
+  uint8_t scope;
+static constexpr auto scope_it_nexus_pos {5u};
+static constexpr uint8_t scope_it_nexus_mask {7u << scope_it_nexus_pos};
+static constexpr auto scope_encryption_pos {0u};
+static constexpr uint8_t scope_encryption_mask {7u << scope_encryption_pos};
+  uint8_t encryption_mode;
+  uint8_t decryption_mode;
```

Comment:
are these modes magic numbers? Do we want an enum here?

Reply:
I added `encrypt_mode` and `decrypt_mode` enums. I think enums for other fields are unnecessary since stenc mostly uses the defaults.